doc/vrf/no-default-vrf-route-fallback-hld.md

```diff
@@ -0,0 +1,411 @@
+                                                        │
+                                                        ▼
+                                               ┌─────────────────┐
+                                               │                 │
+                                               │  Linux Kernel   │
+                                               │ (Unreachable    │
+                                               │  Routes)        │
+                                               └─────────────────┘
+```
+
+## 7. High-Level Design
+
+### 7.1. Design Principles
+
+- **Built-in SONiC Feature**: This is a core SONiC feature, not an Application Extension
+- **Minimal Changes**: Leverages existing infrastructure with minimal modifications
+- **Backward Compatibility**: Maintains current behavior unless explicitly configured
+- **Kernel-Only Solution**: Uses Linux kernel unreachable routes, no hardware programming
+
+### 7.2. Repositories Modified
+
+- **sonic-utilities**: CLI command implementation
+- **sonic-swss**: VRF Manager daemon modifications
+- **sonic-buildimage**: Build configuration updates
+
+### 7.3. Module Dependencies
+
+- **vrfmgrd**: Core module responsible for applying configuration
+- **CONFIG_DB**: Configuration storage
+- **Linux iproute2**: For kernel route manipulation
+
+### 7.4. DB Schema Changes
+
+#### 7.4.1. CONFIG_DB Changes
+
+**New Table: VRF_ROUTE_FALLBACK_GLOBAL**
+```json
+"VRF_ROUTE_FALLBACK_GLOBAL|ipv4": {
+    "status": "enabled"
+},
+"VRF_ROUTE_FALLBACK_GLOBAL|ipv6": {
+    "status": "disabled"
+}
+```
+
+**Extended Table: VRF**
+```json
+"VRF|VrfRed": {
+    "fallback": "enabled",
+    "fallback_v6": "disabled"
+}
+```
+
+### 7.5. Component Design
+
+#### 7.5.1. VRF Manager Changes
+
+**Responsibilities:**
+- Monitor `VRF_ROUTE_FALLBACK_GLOBAL` and `VRF` tables
+- Apply unreachable routes when fallback is disabled
+- Remove unreachable routes when fallback is enabled
+- Handle configuration validation and error cases
+
+**Route Management:**
+- Add: `ip route add unreachable default metric 4278198272 vrf <VRF_NAME>`
+- Remove: `ip route del unreachable default vrf <VRF_NAME>`
+
+#### 7.5.2. CLI Implementation
+
+**Configuration Commands:**
+- `config vrf route-fallback [ipv4|ipv6] [enable|disable]`
+- `config vrf route-fallback [ipv4|ipv6] [enable|disable] -v <VRF_NAME>`
+
+**Show Commands:**
+- `show vrf-route-fallback`
+
+### 7.6. Warm Reboot Impact
+
+- Configuration persists across warm reboots
+- Route reconciliation occurs during vrfmgrd startup
+- No impact on warm reboot timing or data plane disruption
+
+### 7.7. Fastboot Impact
+
+- No additional dependencies for fastboot
+- Configuration loaded during normal startup sequence
+
+## 8. SAI API
+
+**No SAI API changes are required for this feature.**
+
+The unreachable routes are maintained only in the Linux kernel and are explicitly not programmed into the ASIC hardware. This design choice ensures:
```

Comment:
This may not be true. Consider the below config:
```
"VRF_ROUTE_FALLBACK_GLOBAL|ipv4": {
    "status": "enabled"
},
"VRF|VrfRed": {
    "fallback": "enabled",
},
"VRF|VrfGreen": {
    "fallback": "disabled",
}
```
In this case, in the ASIC route tables, **VrfGreen|0.0.0.0/0** route needs to be installed to avoid fallback VRF lookup for packets in VrfGreen.

Reply:
Can you please explain what is the relation of the proposed feature with SAI_SWITCH_ATTR_DEFAULT_VIRTUAL_ROUTER_ID